# Imports dependencies

In [1]:
import numpy as np
import pandas as pd
import os

import tensorflow as tf
import tensorflow_decision_forests as tfdf

print(f"Found TF-DF {tfdf.__version__}")

Found TF-DF 1.9.1


# Load dataset

In [2]:
train_df = pd.read_csv("/kaggle/input/titanic/train.csv")
serving_df = pd.read_csv("/kaggle/input/titanic/test.csv")

train_df.head(10)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   
5            6         0       3   
6            7         0       1   
7            8         0       3   
8            9         1       3   
9           10         1       2   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   
3       Futrelle, Mrs. Jacques Heath (Lily May Peel)  female  35.0      1   
4                           Allen, Mr. William Henry    male  35.0      0   
5                                   Moran, Mr. James    male   NaN      0   
6                            McCarthy, Mr. Timothy J    male  54.0      0   
7                     Palsson, Master. Gosta Leonard    male   2.0      3   
8  Johnson, Mrs. Oscar W (Elisabeth Vilhelmina Berg)  female  27.0      0   
9                Nasser, Mrs. Nicholas (Adele Achem)  female  14.0      1   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S  
3      0            113803  53.1000  C123        S  
4      0            373450   8.0500   NaN        S  
5      0            330877   8.4583   NaN        Q  
6      0             17463  51.8625   E46        S  
7      1            349909  21.0750   NaN        S  
8      2            347742  11.1333   NaN        S  
9      0            237736  30.0708   NaN        C

# Prepare dataset

We will apply the following transformations on the dataset.

1. Tokenize the names. For example, "Braund, Mr. Owen Harris" will become ["Braund", "Mr.", "Owen", "Harris"].
2. Extract any prefix in the ticket. For example ticket "STON/O2. 3101282" will become "STON/O2." and 3101282.

In [3]:
def preprocess(df):
    df = df.copy()
    
    def normalize_name(x):
        return " ".join([v.strip(",()[].\"'") for v in x.split(" ")])
    
    def ticket_number(x):
        return x.split(" ")[-1]
        
    def ticket_item(x):
        items = x.split(" ")
        if len(items) == 1:
            return "NONE"
        return "_".join(items[0:-1])
    
    df["Name"] = df["Name"].apply(normalize_name)
    df["Ticket_number"] = df["Ticket"].apply(ticket_number)
    df["Ticket_item"] = df["Ticket"].apply(ticket_item)                     
    return df
    
preprocessed_train_df = preprocess(train_df)
preprocessed_serving_df = preprocess(serving_df)

preprocessed_train_df.head(5)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   
3            4         1       1   
4            5         0       3   

                                              Name     Sex   Age  SibSp  \
0                            Braund Mr Owen Harris    male  22.0      1   
1  Cumings Mrs John Bradley Florence Briggs Thayer  female  38.0      1   
2                             Heikkinen Miss Laina  female  26.0      0   
3         Futrelle Mrs Jacques Heath Lily May Peel  female  35.0      1   
4                           Allen Mr William Henry    male  35.0      0   

   Parch            Ticket     Fare Cabin Embarked Ticket_number Ticket_item  
0      0         A/5 21171   7.2500   NaN        S         21171         A/5  
1      0          PC 17599  71.2833   C85        C         17599          PC  
2      0  STON/O2. 3101282   7.9250   NaN        S       3101282    STON/O2.  
3      0            113803  53.1000  C123        S        113803        NONE  
4      0            373450   8.0500   NaN        S        373450        NONE

Let's keep the list of the input features of the model. Notably, we don't want to train our model on the "PassengerId" and "Ticket" features.

In [4]:
input_features = list(preprocessed_train_df.columns)
input_features.remove("Ticket")
input_features.remove("PassengerId")
input_features.remove("Survived")
#input_features.remove("Ticket_number")

print(f"Input features: {input_features}")

Input features: ['Pclass', 'Name', 'Sex', 'Age', 'SibSp', 'Parch', 'Fare', 'Cabin', 'Embarked', 'Ticket_number', 'Ticket_item']


# Convert Pandas dataset to TensorFlow Dataset

In [5]:
def tokenize_names(features, labels=None):
    """Divite the names into tokens. TF-DF can consume text tokens natively."""
    features["Name"] =  tf.strings.split(features["Name"])
    return features, labels

train_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_train_df,label="Survived").map(tokenize_names)
serving_ds = tfdf.keras.pd_dataframe_to_tf_dataset(preprocessed_serving_df).map(tokenize_names)

# Train model with default parameters

### Train model

First, we are training a GradientBoostedTreesModel model with the default parameters.

In [6]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    random_seed=1234,
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 24-12-21 20:48:41.1088 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:41.1098 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:41.1098 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:48:45.9810 UTC kernel.cc:1233] Loading model from path /tmp/tmp77omi7pn/model/ with prefix 160982ff62234627
[INFO 24-12-21 20:48:45.9873 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-21 20:48:45.9878 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:48:45.9878 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.8260869383811951 Loss:0.8608942627906799


# Train model with improved default parameters

Now you'll use some specific parameters when creating the GBT model

In [7]:
model = tfdf.keras.GradientBoostedTreesModel(
    verbose=0, # Very few logs
    features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
    exclude_non_specified_features=True, # Only use the features in "features"
    
    #num_trees=2000,
    
    # Only for GBT.
    # A bit slower, but great to understand the model.
    # compute_permutation_variable_importance=True,
    
    # Change the default hyper-parameters
    # hyperparameter_template="benchmark_rank1@v1",
    
    #num_trees=1000,
    #tuner=tuner
    
    min_examples=1,
    categorical_algorithm="RANDOM",
    #max_depth=4,
    shrinkage=0.05,
    #num_candidate_attributes_ratio=0.2,
    split_axis="SPARSE_OBLIQUE",
    sparse_oblique_normalization="MIN_MAX",
    sparse_oblique_num_projections_exponent=2.0,
    num_trees=2000,
    #validation_ratio=0.0,
    random_seed=1234,
    
)
model.fit(train_ds)

self_evaluation = model.make_inspector().evaluation()
print(f"Accuracy: {self_evaluation.accuracy} Loss:{self_evaluation.loss}")

[WARNING 24-12-21 20:48:51.5560 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:51.5560 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:51.5561 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:48:52.4422 UTC kernel.cc:1233] Loading model from path /tmp/tmp6wjkht2v/model/ with prefix 30c902e96a694a35
[INFO 24-12-21 20:48:52.4512 UTC decision_forest.cc:734] Model loaded with 40 root(s), 2106 node(s), and 10 input feature(s).
[INFO 24-12-21 20:48:52.4513 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.782608687877655 Loss:1.0586705207824707


Let's look at the model and you can also notice the information about variable importance that the model figured out

In [8]:
model.summary()

Model: "gradient_boosted_trees_model_1"
_________________________________________________________________
 Layer (type)                Output Shape              Param #   
Total params: 1 (1.00 Byte)
Trainable params: 0 (0.00 Byte)
Non-trainable params: 1 (1.00 Byte)
_________________________________________________________________
Type: "GRADIENT_BOOSTED_TREES"
Task: CLASSIFICATION
Label: "__LABEL"

Input Features (11):
	Age
	Cabin
	Embarked
	Fare
	Name
	Parch
	Pclass
	Sex
	SibSp
	Ticket_item
	Ticket_number

No weights

Variable Importance: INV_MEAN_MIN_DEPTH:
    1.           "Sex"  0.585997 ################
    2.           "Age"  0.364636 #######
    3.          "Fare"  0.266191 ###
    4.          "Name"  0.207054 #
    5.        "Pclass"  0.179191 
    6. "Ticket_number"  0.178806 
    7.      "Embarked"  0.177803 
    8.   "Ticket_item"  0.177009 
    9.         "Parch"  0.175276 
   10.         "SibSp"  0.171694 

Variable Importance: NUM_AS_ROOT:
    1.  "Sex" 34.000000 ######

# Make predictions

In [9]:
def prediction_to_kaggle_format(model, threshold=0.5):
    proba_survive = model.predict(serving_ds, verbose=0)[:,0]
    return pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (proba_survive >= threshold).astype(int)
    })

def make_submission(kaggle_predictions):
    path="/kaggle/working/submission.csv"
    kaggle_predictions.to_csv(path, index=False)
    print(f"Submission exported to {path}")
    
kaggle_predictions = prediction_to_kaggle_format(model)
make_submission(kaggle_predictions)
!head /kaggle/working/submission.csv

Submission exported to /kaggle/working/submission.csv
PassengerId,Survived
892,0
893,0
894,0
895,0
896,0
897,0
898,0
899,0
900,1


# Training a model with hyperparameter tunning

Hyper-parameter tuning is enabled by specifying the tuner constructor argument of the model. The tuner object contains all the configuration of the tuner (search space, optimizer, trial and objective).


In [10]:
tuner = tfdf.tuner.RandomSearch(num_trials=1000)
tuner.choice("min_examples", [2, 5, 7, 10])
tuner.choice("categorical_algorithm", ["CART", "RANDOM"])

local_search_space = tuner.choice("growing_strategy", ["LOCAL"])
local_search_space.choice("max_depth", [3, 4, 5, 6, 8])

global_search_space = tuner.choice("growing_strategy", ["BEST_FIRST_GLOBAL"], merge=True)
global_search_space.choice("max_num_nodes", [16, 32, 64, 128, 256])

#tuner.choice("use_hessian_gain", [True, False])
tuner.choice("shrinkage", [0.02, 0.05, 0.10, 0.15])
tuner.choice("num_candidate_attributes_ratio", [0.2, 0.5, 0.9, 1.0])


tuner.choice("split_axis", ["AXIS_ALIGNED"])
oblique_space = tuner.choice("split_axis", ["SPARSE_OBLIQUE"], merge=True)
oblique_space.choice("sparse_oblique_normalization",
                     ["NONE", "STANDARD_DEVIATION", "MIN_MAX"])
oblique_space.choice("sparse_oblique_weights", ["BINARY", "CONTINUOUS"])
oblique_space.choice("sparse_oblique_num_projections_exponent", [1.0, 1.5])

# Tune the model. Notice the `tuner=tuner`.
tuned_model = tfdf.keras.GradientBoostedTreesModel(tuner=tuner)
tuned_model.fit(train_ds, verbose=0)

tuned_self_evaluation = tuned_model.make_inspector().evaluation()
print(f"Accuracy: {tuned_self_evaluation.accuracy} Loss:{tuned_self_evaluation.loss}")

Use /tmp/tmplwi44gad as temporary training directory


[WARNING 24-12-21 20:48:54.2310 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:54.2310 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:48:54.2310 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:27.5226 UTC kernel.cc:1233] Loading model from path /tmp/tmplwi44gad/model/ with prefix 8a12a09d4b9a4228
[INFO 24-12-21 20:54:27.5442 UTC decision_forest.cc:734] Model loaded with 30 root(s), 2236 node(s), and 12 input feature(s).
[INFO 24-12-21 20:54:27.5442 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesGeneric" built
[INFO 24-12-21 20:54:27.5443 UTC kernel.cc:1061] Use fast generic engine


Accuracy: 0.8630136847496033 Loss:0.6749962568283081


In the last line in the cell above, you can see the accuracy is higher than previously with default parameters and parameters set by hand.

This is the main idea behing hyperparameter tuning.

For more information you can follow this tutorial: [Automated hyper-parameter tuning](https://www.tensorflow.org/decision_forests/tutorials/automatic_tuning_colab)

# Making an ensemble

Here you'll create 100 models with different seeds and combine their results

This approach removes a little bit the random aspects related to creating ML models

In the GBT creation is used the `honest` parameter. It will use different training examples to infer the structure and the leaf values. This regularization technique trades examples for bias estimates.

In [11]:
predictions = None
num_predictions = 0

for i in range(100):
    print(f"i:{i}")
    # Possible models: GradientBoostedTreesModel or RandomForestModel
    model = tfdf.keras.GradientBoostedTreesModel(
        verbose=0, # Very few logs
        features=[tfdf.keras.FeatureUsage(name=n) for n in input_features],
        exclude_non_specified_features=True, # Only use the features in "features"

        #min_examples=1,
        #categorical_algorithm="RANDOM",
        ##max_depth=4,
        #shrinkage=0.05,
        ##num_candidate_attributes_ratio=0.2,
        #split_axis="SPARSE_OBLIQUE",
        #sparse_oblique_normalization="MIN_MAX",
        #sparse_oblique_num_projections_exponent=2.0,
        #num_trees=2000,
        ##validation_ratio=0.0,
        random_seed=i,
        honest=True,
    )
    model.fit(train_ds)
    
    sub_predictions = model.predict(serving_ds, verbose=0)[:,0]
    if predictions is None:
        predictions = sub_predictions
    else:
        predictions += sub_predictions
    num_predictions += 1

predictions/=num_predictions

kaggle_predictions = pd.DataFrame({
        "PassengerId": serving_df["PassengerId"],
        "Survived": (predictions >= 0.5).astype(int)
    })

make_submission(kaggle_predictions)

i:0


[WARNING 24-12-21 20:54:27.9688 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:27.9688 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:27.9689 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:28.4517 UTC kernel.cc:1233] Loading model from path /tmp/tmp9_fltmfj/model/ with prefix abf2604d831848f2
[INFO 24-12-21 20:54:28.4561 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-21 20:54:28.4564 UTC kernel.cc:1061] Use fast generic engine


i:1


[WARNING 24-12-21 20:54:28.8834 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:28.8834 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:28.8834 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:29.7658 UTC kernel.cc:1233] Loading model from path /tmp/tmp7dq5gsiu/model/ with prefix c61c4b09f9c54d64
[INFO 24-12-21 20:54:29.7866 UTC kernel.cc:1061] Use fast generic engine


i:2


[WARNING 24-12-21 20:54:30.2243 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:30.2243 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:30.2243 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:30.7379 UTC kernel.cc:1233] Loading model from path /tmp/tmp630tq2zp/model/ with prefix 2eadbab9183044a9
[INFO 24-12-21 20:54:30.7428 UTC kernel.cc:1061] Use fast generic engine


i:3


[WARNING 24-12-21 20:54:31.1784 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:31.1784 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:31.1784 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:32.2459 UTC kernel.cc:1233] Loading model from path /tmp/tmpfqolt9rq/model/ with prefix 919edb61e66a4d06
[INFO 24-12-21 20:54:32.2734 UTC kernel.cc:1061] Use fast generic engine


i:4


[WARNING 24-12-21 20:54:32.7049 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:32.7049 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:32.7049 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:33.2437 UTC kernel.cc:1233] Loading model from path /tmp/tmpygnvfezv/model/ with prefix f79f85019a19494e
[INFO 24-12-21 20:54:33.2495 UTC kernel.cc:1061] Use fast generic engine


i:5


[WARNING 24-12-21 20:54:33.6810 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:33.6810 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:33.6811 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:34.1590 UTC kernel.cc:1233] Loading model from path /tmp/tmpuc5kn6ls/model/ with prefix 4725c07ad60c4334
[INFO 24-12-21 20:54:34.1625 UTC kernel.cc:1061] Use fast generic engine


i:6


[WARNING 24-12-21 20:54:34.5896 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:34.5896 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:34.5896 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:35.1525 UTC kernel.cc:1233] Loading model from path /tmp/tmpeqa7vkov/model/ with prefix 0a6a3ef4af294ab8
[INFO 24-12-21 20:54:35.1605 UTC kernel.cc:1061] Use fast generic engine


i:7


[WARNING 24-12-21 20:54:35.5795 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:35.5796 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:35.5796 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:36.5189 UTC kernel.cc:1233] Loading model from path /tmp/tmp8qssfte5/model/ with prefix 3f4dc25308884153
[INFO 24-12-21 20:54:36.5410 UTC kernel.cc:1061] Use fast generic engine


i:8


[WARNING 24-12-21 20:54:37.0374 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:37.0375 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:37.0375 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:37.7612 UTC kernel.cc:1233] Loading model from path /tmp/tmp9l87rbkb/model/ with prefix ef540144ec274ea8
[INFO 24-12-21 20:54:37.7716 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:54:37.7717 UTC kernel.cc:1061] Use fast generic engine


i:9


[WARNING 24-12-21 20:54:38.2124 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:38.2125 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:38.2125 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:38.9845 UTC kernel.cc:1233] Loading model from path /tmp/tmpz1muc909/model/ with prefix af3713d8e16e44e6
[INFO 24-12-21 20:54:39.0002 UTC kernel.cc:1061] Use fast generic engine


i:10


[WARNING 24-12-21 20:54:39.4386 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:39.4387 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:39.4387 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:39.9811 UTC kernel.cc:1233] Loading model from path /tmp/tmpg0xcwxo4/model/ with prefix a30a383444f14598
[INFO 24-12-21 20:54:39.9874 UTC kernel.cc:1061] Use fast generic engine


i:11


[WARNING 24-12-21 20:54:40.4163 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:40.4163 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:40.4163 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:41.1719 UTC kernel.cc:1233] Loading model from path /tmp/tmpmc17_cdb/model/ with prefix 6252baacfbda466b
[INFO 24-12-21 20:54:41.1869 UTC kernel.cc:1061] Use fast generic engine


i:12


[WARNING 24-12-21 20:54:41.6169 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:41.6170 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:41.6170 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:42.1669 UTC kernel.cc:1233] Loading model from path /tmp/tmpg9n80c8h/model/ with prefix d3170816c2ed41a9
[INFO 24-12-21 20:54:42.1731 UTC kernel.cc:1061] Use fast generic engine


i:13


[WARNING 24-12-21 20:54:42.6056 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:42.6056 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:42.6056 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:43.7381 UTC kernel.cc:1233] Loading model from path /tmp/tmpjrpg6a5x/model/ with prefix 1537aa154e084be0
[INFO 24-12-21 20:54:43.7518 UTC kernel.cc:1061] Use fast generic engine


i:14


[WARNING 24-12-21 20:54:44.2118 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:44.2118 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:44.2118 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:44.7601 UTC kernel.cc:1233] Loading model from path /tmp/tmpx1dtqs2s/model/ with prefix 32e49c379dd941c8
[INFO 24-12-21 20:54:44.7674 UTC kernel.cc:1061] Use fast generic engine


i:15


[WARNING 24-12-21 20:54:45.2135 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:45.2135 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:45.2136 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:45.7850 UTC kernel.cc:1233] Loading model from path /tmp/tmpw2amy_gx/model/ with prefix 61271449d6374bb9
[INFO 24-12-21 20:54:45.7934 UTC kernel.cc:1061] Use fast generic engine


i:16


[WARNING 24-12-21 20:54:46.2303 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:46.2303 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:46.2303 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:46.9634 UTC kernel.cc:1233] Loading model from path /tmp/tmp6r_px4e4/model/ with prefix 3cc1a901494f41c8
[INFO 24-12-21 20:54:46.9832 UTC kernel.cc:1061] Use fast generic engine


i:17


[WARNING 24-12-21 20:54:47.4233 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:47.4233 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:47.4233 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:48.1441 UTC kernel.cc:1233] Loading model from path /tmp/tmpuonyrr08/model/ with prefix d2a3fee4446c4a09
[INFO 24-12-21 20:54:48.1582 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:54:48.1582 UTC kernel.cc:1061] Use fast generic engine


i:18


[WARNING 24-12-21 20:54:48.5806 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:48.5807 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:48.5807 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:49.2684 UTC kernel.cc:1233] Loading model from path /tmp/tmp1pa284en/model/ with prefix 5499aaf687334fe6
[INFO 24-12-21 20:54:49.2812 UTC kernel.cc:1061] Use fast generic engine


i:19


[WARNING 24-12-21 20:54:49.7078 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:49.7078 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:49.7078 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:50.5810 UTC kernel.cc:1233] Loading model from path /tmp/tmpiis6k5qm/model/ with prefix 8ebb604876e84e2c
[INFO 24-12-21 20:54:50.6000 UTC kernel.cc:1061] Use fast generic engine


i:20


[WARNING 24-12-21 20:54:51.0386 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:51.0387 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:51.0387 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:51.8234 UTC kernel.cc:1233] Loading model from path /tmp/tmpjcof9rnh/model/ with prefix d8e4a5f65d1c4b67
[INFO 24-12-21 20:54:51.8391 UTC kernel.cc:1061] Use fast generic engine


i:21


[WARNING 24-12-21 20:54:52.2739 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:52.2740 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:52.2740 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:52.8029 UTC kernel.cc:1233] Loading model from path /tmp/tmpx7e2u1ko/model/ with prefix 032458fa0ea64291
[INFO 24-12-21 20:54:52.8090 UTC kernel.cc:1061] Use fast generic engine


i:22


[WARNING 24-12-21 20:54:53.2414 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:53.2415 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:53.2415 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:53.7813 UTC kernel.cc:1233] Loading model from path /tmp/tmpjf188jva/model/ with prefix 54fccfd47a8e44c5
[INFO 24-12-21 20:54:53.7874 UTC kernel.cc:1061] Use fast generic engine


i:23


[WARNING 24-12-21 20:54:54.2202 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:54.2202 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:54.2202 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:54.8268 UTC kernel.cc:1233] Loading model from path /tmp/tmpddi3_c1a/model/ with prefix 91f44824fa9344c6
[INFO 24-12-21 20:54:54.8363 UTC kernel.cc:1061] Use fast generic engine


i:24


[WARNING 24-12-21 20:54:55.2705 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:55.2705 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:55.2705 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:55.8132 UTC kernel.cc:1233] Loading model from path /tmp/tmptv3vlj3_/model/ with prefix 7d1b5fbff32e4a5a
[INFO 24-12-21 20:54:55.8194 UTC kernel.cc:1061] Use fast generic engine


i:25


[WARNING 24-12-21 20:54:56.2534 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:56.2535 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:56.2535 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:56.9469 UTC kernel.cc:1233] Loading model from path /tmp/tmp8pshgzt4/model/ with prefix 2eb9cb31d29f4c6a
[INFO 24-12-21 20:54:56.9610 UTC kernel.cc:1061] Use fast generic engine


i:26


[WARNING 24-12-21 20:54:57.4100 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:57.4101 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:57.4101 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:58.0609 UTC kernel.cc:1233] Loading model from path /tmp/tmpdc21a4xr/model/ with prefix 98f05538315644b5
[INFO 24-12-21 20:54:58.0716 UTC kernel.cc:1061] Use fast generic engine


i:27


[WARNING 24-12-21 20:54:58.4945 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:58.4945 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:58.4946 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:54:59.0433 UTC kernel.cc:1233] Loading model from path /tmp/tmpk5nk441z/model/ with prefix fefc98c0efa1420c
[INFO 24-12-21 20:54:59.0502 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-21 20:54:59.0506 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:54:59.0506 UTC kernel.cc:1061] Use fast generic engine


i:28


[WARNING 24-12-21 20:54:59.4776 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:59.4776 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:54:59.4776 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:00.0145 UTC kernel.cc:1233] Loading model from path /tmp/tmpu9b9qje7/model/ with prefix e1b2b860ccc84086
[INFO 24-12-21 20:55:00.0201 UTC kernel.cc:1061] Use fast generic engine


i:29


[WARNING 24-12-21 20:55:00.4524 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:00.4524 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:00.4524 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:01.1878 UTC kernel.cc:1233] Loading model from path /tmp/tmploehlgm7/model/ with prefix aa51960e77ba431b
[INFO 24-12-21 20:55:01.2014 UTC kernel.cc:1061] Use fast generic engine


i:30


[WARNING 24-12-21 20:55:01.6320 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:01.6320 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:01.6320 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:02.7311 UTC kernel.cc:1233] Loading model from path /tmp/tmp6qdzczcq/model/ with prefix c3ac02e1b4894715
[INFO 24-12-21 20:55:02.7610 UTC kernel.cc:1061] Use fast generic engine


i:31


[WARNING 24-12-21 20:55:03.1977 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:03.1977 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:03.1977 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:03.8348 UTC kernel.cc:1233] Loading model from path /tmp/tmp6aaa4_z0/model/ with prefix 00260fdc53864acd
[INFO 24-12-21 20:55:03.8448 UTC kernel.cc:1061] Use fast generic engine


i:32


[WARNING 24-12-21 20:55:04.2749 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:04.2749 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:04.2749 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:04.8125 UTC kernel.cc:1233] Loading model from path /tmp/tmpux1b6vo3/model/ with prefix 8a27d9a88ab24e3d
[INFO 24-12-21 20:55:04.8185 UTC kernel.cc:1061] Use fast generic engine


i:33


[WARNING 24-12-21 20:55:05.2450 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:05.2450 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:05.2450 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:05.9845 UTC kernel.cc:1233] Loading model from path /tmp/tmpi_we574a/model/ with prefix fd7881d15ef3470f
[INFO 24-12-21 20:55:05.9984 UTC kernel.cc:1061] Use fast generic engine


i:34


[WARNING 24-12-21 20:55:06.4290 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:06.4290 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:06.4290 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:07.0618 UTC kernel.cc:1233] Loading model from path /tmp/tmpk98wjw7r/model/ with prefix d75bccc5f3fd414e
[INFO 24-12-21 20:55:07.0710 UTC kernel.cc:1061] Use fast generic engine


i:35


[WARNING 24-12-21 20:55:07.4983 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:07.4983 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:07.4983 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:08.6054 UTC kernel.cc:1233] Loading model from path /tmp/tmpa5xps_s5/model/ with prefix 88d5839bcc7740d3
[INFO 24-12-21 20:55:08.6157 UTC kernel.cc:1061] Use fast generic engine


i:36


[WARNING 24-12-21 20:55:09.1298 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:09.1299 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:09.1299 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:09.9685 UTC kernel.cc:1233] Loading model from path /tmp/tmprm_8e9a7/model/ with prefix adeee5bd89324431
[INFO 24-12-21 20:55:09.9867 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:55:09.9868 UTC kernel.cc:1061] Use fast generic engine


i:37


[WARNING 24-12-21 20:55:10.4527 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:10.4528 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:10.4528 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:11.0879 UTC kernel.cc:1233] Loading model from path /tmp/tmpq0mrmwgk/model/ with prefix c14baf5e8f094d6c
[INFO 24-12-21 20:55:11.0984 UTC kernel.cc:1061] Use fast generic engine


i:38


[WARNING 24-12-21 20:55:11.5522 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:11.5522 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:11.5522 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:12.3046 UTC kernel.cc:1233] Loading model from path /tmp/tmps7asvqwt/model/ with prefix 0065824614f941e1
[INFO 24-12-21 20:55:12.3201 UTC kernel.cc:1061] Use fast generic engine


i:39


[WARNING 24-12-21 20:55:12.7676 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:12.7676 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:12.7676 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:13.5095 UTC kernel.cc:1233] Loading model from path /tmp/tmpkdo9f9m5/model/ with prefix 6eff56d7feff47c0
[INFO 24-12-21 20:55:13.5240 UTC kernel.cc:1061] Use fast generic engine


i:40


[WARNING 24-12-21 20:55:13.9644 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:13.9644 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:13.9644 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:14.4712 UTC kernel.cc:1233] Loading model from path /tmp/tmpqyco9rd8/model/ with prefix eff9f78711ea43b8
[INFO 24-12-21 20:55:14.4760 UTC kernel.cc:1061] Use fast generic engine


i:41


[WARNING 24-12-21 20:55:14.9039 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:14.9039 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:14.9039 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:15.6875 UTC kernel.cc:1233] Loading model from path /tmp/tmp3z6g8uyw/model/ with prefix 264a7f58073f41ce
[INFO 24-12-21 20:55:15.7041 UTC kernel.cc:1061] Use fast generic engine


i:42


[WARNING 24-12-21 20:55:16.1400 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:16.1401 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:16.1401 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:16.7768 UTC kernel.cc:1233] Loading model from path /tmp/tmpuwrapthe/model/ with prefix 0b8f4016699c4792
[INFO 24-12-21 20:55:16.7887 UTC kernel.cc:1061] Use fast generic engine


i:43


[WARNING 24-12-21 20:55:17.2413 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:17.2414 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:17.2414 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:18.0798 UTC kernel.cc:1233] Loading model from path /tmp/tmp4_u9y9z3/model/ with prefix 53e4a1350fb54f0d
[INFO 24-12-21 20:55:18.0979 UTC kernel.cc:1061] Use fast generic engine


i:44


[WARNING 24-12-21 20:55:18.5242 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:18.5243 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:18.5243 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:19.1494 UTC kernel.cc:1233] Loading model from path /tmp/tmpdkieyqrq/model/ with prefix 2ea3eabd60dc4b5b
[INFO 24-12-21 20:55:19.1595 UTC kernel.cc:1061] Use fast generic engine


i:45


[WARNING 24-12-21 20:55:19.5929 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:19.5930 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:19.5930 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:20.0742 UTC kernel.cc:1233] Loading model from path /tmp/tmpywtor_l9/model/ with prefix f5b94fa4828541b3
[INFO 24-12-21 20:55:20.0778 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:55:20.0779 UTC kernel.cc:1061] Use fast generic engine


i:46


[WARNING 24-12-21 20:55:20.5069 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:20.5070 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:20.5070 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:21.2882 UTC kernel.cc:1233] Loading model from path /tmp/tmp0scj8hzx/model/ with prefix 76689d7c9cfd403c
[INFO 24-12-21 20:55:21.3032 UTC kernel.cc:1061] Use fast generic engine


i:47


[WARNING 24-12-21 20:55:21.7291 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:21.7291 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:21.7291 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:22.4822 UTC kernel.cc:1233] Loading model from path /tmp/tmp002ok5so/model/ with prefix 0232d75d7e9b44c3
[INFO 24-12-21 20:55:22.4960 UTC kernel.cc:1061] Use fast generic engine


i:48


[WARNING 24-12-21 20:55:22.9355 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:22.9356 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:22.9356 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:23.4537 UTC kernel.cc:1233] Loading model from path /tmp/tmpkzte3wfu/model/ with prefix 1adaed2fd0ec4329
[INFO 24-12-21 20:55:23.4585 UTC kernel.cc:1061] Use fast generic engine


i:49


[WARNING 24-12-21 20:55:23.8826 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:23.8827 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:23.8827 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:24.4632 UTC kernel.cc:1233] Loading model from path /tmp/tmpbc6cnpys/model/ with prefix 6a1c141f8aa540ec
[INFO 24-12-21 20:55:24.4704 UTC kernel.cc:1061] Use fast generic engine


i:50


[WARNING 24-12-21 20:55:24.8969 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:24.8969 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:24.8969 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:25.6044 UTC kernel.cc:1233] Loading model from path /tmp/tmptgmnwlsg/model/ with prefix dea2773ea2334ec8
[INFO 24-12-21 20:55:25.6173 UTC kernel.cc:1061] Use fast generic engine


i:51


[WARNING 24-12-21 20:55:26.0488 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:26.0489 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:26.0489 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:26.8555 UTC kernel.cc:1233] Loading model from path /tmp/tmpgh0h362g/model/ with prefix 024697ed960f4c34
[INFO 24-12-21 20:55:26.8716 UTC kernel.cc:1061] Use fast generic engine


i:52


[WARNING 24-12-21 20:55:27.3296 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:27.3297 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:27.3297 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:27.9120 UTC kernel.cc:1233] Loading model from path /tmp/tmpuxvx7ous/model/ with prefix a82e524d68a84f91
[INFO 24-12-21 20:55:27.9205 UTC kernel.cc:1061] Use fast generic engine


i:53


[WARNING 24-12-21 20:55:28.3495 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:28.3495 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:28.3496 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:28.9281 UTC kernel.cc:1233] Loading model from path /tmp/tmpei0064dj/model/ with prefix 6cccc81d051e44d4
[INFO 24-12-21 20:55:28.9360 UTC kernel.cc:1061] Use fast generic engine


i:54


[WARNING 24-12-21 20:55:29.3690 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:29.3691 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:29.3691 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:29.8495 UTC kernel.cc:1233] Loading model from path /tmp/tmpi6m_k59o/model/ with prefix 343f8dee6fef4631
[INFO 24-12-21 20:55:29.8529 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-21 20:55:29.8532 UTC kernel.cc:1061] Use fast generic engine


i:55


[WARNING 24-12-21 20:55:30.2885 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:30.2885 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:30.2885 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:31.0722 UTC kernel.cc:1233] Loading model from path /tmp/tmp854wajqj/model/ with prefix 43b84d9ce9024594
[INFO 24-12-21 20:55:31.0876 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:55:31.0877 UTC kernel.cc:1061] Use fast generic engine


i:56


[WARNING 24-12-21 20:55:31.5209 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:31.5209 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:31.5209 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:32.1879 UTC kernel.cc:1233] Loading model from path /tmp/tmpux8fpct3/model/ with prefix 937ce90cb37d43d9
[INFO 24-12-21 20:55:32.1993 UTC kernel.cc:1061] Use fast generic engine


i:57


[WARNING 24-12-21 20:55:32.6329 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:32.6329 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:32.6329 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:33.1457 UTC kernel.cc:1233] Loading model from path /tmp/tmp7p90zz48/model/ with prefix 1bf6abc0da0d45f0
[INFO 24-12-21 20:55:33.1500 UTC kernel.cc:1061] Use fast generic engine


i:58


[WARNING 24-12-21 20:55:33.5785 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:33.5786 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:33.5786 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:34.1305 UTC kernel.cc:1233] Loading model from path /tmp/tmphrqyb6u6/model/ with prefix 435638a2b1544c1c
[INFO 24-12-21 20:55:34.1375 UTC kernel.cc:1061] Use fast generic engine


i:59


[WARNING 24-12-21 20:55:34.5662 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:34.5662 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:34.5662 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:35.2009 UTC kernel.cc:1233] Loading model from path /tmp/tmpozopr3wh/model/ with prefix 284a2a97d1e44a14
[INFO 24-12-21 20:55:35.2106 UTC kernel.cc:1061] Use fast generic engine


i:60


[WARNING 24-12-21 20:55:35.6375 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:35.6375 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:35.6375 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:36.2894 UTC kernel.cc:1233] Loading model from path /tmp/tmpev1u75fq/model/ with prefix fa6753b040b54e62
[INFO 24-12-21 20:55:36.2998 UTC kernel.cc:1061] Use fast generic engine


i:61


[WARNING 24-12-21 20:55:36.7289 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:36.7290 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:36.7290 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:37.2902 UTC kernel.cc:1233] Loading model from path /tmp/tmp7qtz675e/model/ with prefix d7e696510d884fb2
[INFO 24-12-21 20:55:37.2955 UTC kernel.cc:1061] Use fast generic engine


i:62


[WARNING 24-12-21 20:55:37.7281 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:37.7282 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:37.7282 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:39.1113 UTC kernel.cc:1233] Loading model from path /tmp/tmp1ivxt5oh/model/ with prefix 9cc929acb3144b13
[INFO 24-12-21 20:55:39.1290 UTC kernel.cc:1061] Use fast generic engine


i:63


[WARNING 24-12-21 20:55:39.6302 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:39.6303 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:39.6303 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:40.2787 UTC kernel.cc:1233] Loading model from path /tmp/tmp04igyfqu/model/ with prefix 2eaed5b22f5d4298
[INFO 24-12-21 20:55:40.2901 UTC kernel.cc:1061] Use fast generic engine


i:64


[WARNING 24-12-21 20:55:40.7816 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:40.7816 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:40.7816 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:41.4352 UTC kernel.cc:1233] Loading model from path /tmp/tmpdwfw04mq/model/ with prefix c08920b337b44836
[INFO 24-12-21 20:55:41.4449 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:55:41.4449 UTC kernel.cc:1061] Use fast generic engine


i:65


[WARNING 24-12-21 20:55:41.9120 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:41.9120 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:41.9120 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:42.4401 UTC kernel.cc:1233] Loading model from path /tmp/tmp4jzpmnpw/model/ with prefix d1d2a0f302ad4a1d
[INFO 24-12-21 20:55:42.4457 UTC kernel.cc:1061] Use fast generic engine


i:66


[WARNING 24-12-21 20:55:42.9015 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:42.9015 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:42.9015 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:43.4697 UTC kernel.cc:1233] Loading model from path /tmp/tmpifdjno0p/model/ with prefix fea6d6610f96439f
[INFO 24-12-21 20:55:43.4768 UTC kernel.cc:1061] Use fast generic engine


i:67


[WARNING 24-12-21 20:55:43.9267 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:43.9268 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:43.9268 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:44.8089 UTC kernel.cc:1233] Loading model from path /tmp/tmpy73eba5y/model/ with prefix 9889e3e9077c4ae7
[INFO 24-12-21 20:55:44.8299 UTC kernel.cc:1061] Use fast generic engine


i:68


[WARNING 24-12-21 20:55:45.2786 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:45.2786 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:45.2786 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:45.9603 UTC kernel.cc:1233] Loading model from path /tmp/tmphkhx5te5/model/ with prefix b885d138ec6642d3
[INFO 24-12-21 20:55:45.9732 UTC kernel.cc:1061] Use fast generic engine


i:69


[WARNING 24-12-21 20:55:46.4130 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:46.4131 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:46.4131 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:46.9857 UTC kernel.cc:1233] Loading model from path /tmp/tmp80r52r0f/model/ with prefix 287760b6bacb47d3
[INFO 24-12-21 20:55:46.9937 UTC kernel.cc:1061] Use fast generic engine


i:70


[WARNING 24-12-21 20:55:47.4360 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:47.4361 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:47.4361 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:48.0294 UTC kernel.cc:1233] Loading model from path /tmp/tmpakzta815/model/ with prefix b5d2a48e2e0b4c91
[INFO 24-12-21 20:55:48.0390 UTC kernel.cc:1061] Use fast generic engine


i:71


[WARNING 24-12-21 20:55:48.4645 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:48.4645 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:48.4645 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:49.0247 UTC kernel.cc:1233] Loading model from path /tmp/tmppr59d_d0/model/ with prefix 3814fb6864374df5
[INFO 24-12-21 20:55:49.0327 UTC kernel.cc:1061] Use fast generic engine


i:72


[WARNING 24-12-21 20:55:49.4732 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:49.4733 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:49.4733 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:50.2712 UTC kernel.cc:1233] Loading model from path /tmp/tmphoposaa4/model/ with prefix 437d4fb709e94b34
[INFO 24-12-21 20:55:50.2879 UTC kernel.cc:1061] Use fast generic engine


i:73


[WARNING 24-12-21 20:55:50.7242 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:50.7243 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:50.7243 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:51.2963 UTC kernel.cc:1233] Loading model from path /tmp/tmpbmelt0x3/model/ with prefix 717666e1047844ae
[INFO 24-12-21 20:55:51.3037 UTC kernel.cc:1061] Use fast generic engine


i:74


[WARNING 24-12-21 20:55:51.7338 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:51.7338 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:51.7338 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:52.4433 UTC kernel.cc:1233] Loading model from path /tmp/tmpc6t7g1yd/model/ with prefix ff76b1b0e948430d
[INFO 24-12-21 20:55:52.4556 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:55:52.4556 UTC kernel.cc:1061] Use fast generic engine


i:75


[WARNING 24-12-21 20:55:52.8951 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:52.8951 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:52.8951 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:53.4980 UTC kernel.cc:1233] Loading model from path /tmp/tmpg6yaasak/model/ with prefix d2d2ccdd98104756
[INFO 24-12-21 20:55:53.5065 UTC kernel.cc:1061] Use fast generic engine


i:76


[WARNING 24-12-21 20:55:53.9417 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:53.9418 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:53.9418 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:54.4468 UTC kernel.cc:1233] Loading model from path /tmp/tmp5csvrm17/model/ with prefix 938a3139657846a4
[INFO 24-12-21 20:55:54.4513 UTC kernel.cc:1061] Use fast generic engine


i:77


[WARNING 24-12-21 20:55:54.8791 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:54.8791 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:54.8791 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:55.3822 UTC kernel.cc:1233] Loading model from path /tmp/tmpyt5f7_hn/model/ with prefix 271893c4699e4daa
[INFO 24-12-21 20:55:55.3871 UTC kernel.cc:1061] Use fast generic engine


i:78


[WARNING 24-12-21 20:55:55.8192 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:55.8193 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:55.8193 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:56.3826 UTC kernel.cc:1233] Loading model from path /tmp/tmpwnzmw674/model/ with prefix 1c1a9332b60549b0
[INFO 24-12-21 20:55:56.3897 UTC kernel.cc:1061] Use fast generic engine


i:79


[WARNING 24-12-21 20:55:56.8239 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:56.8239 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:56.8239 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:57.4112 UTC kernel.cc:1233] Loading model from path /tmp/tmphej1uuyh/model/ with prefix 5761b85b1f3d4290
[INFO 24-12-21 20:55:57.4187 UTC kernel.cc:1061] Use fast generic engine


i:80


[WARNING 24-12-21 20:55:57.8508 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:57.8509 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:57.8509 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:58.4887 UTC kernel.cc:1233] Loading model from path /tmp/tmpf3fcjr0_/model/ with prefix 5c542ad6380242eb
[INFO 24-12-21 20:55:58.4990 UTC kernel.cc:1061] Use fast generic engine


i:81


[WARNING 24-12-21 20:55:58.9244 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:58.9244 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:55:58.9245 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:55:59.6123 UTC kernel.cc:1233] Loading model from path /tmp/tmpamqvqprg/model/ with prefix faa0377dbc1241e3
[INFO 24-12-21 20:55:59.6239 UTC kernel.cc:1061] Use fast generic engine


i:82


[WARNING 24-12-21 20:56:00.0558 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:00.0558 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:00.0558 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:00.6988 UTC kernel.cc:1233] Loading model from path /tmp/tmpjhkhyif6/model/ with prefix 427e54d55ea040c7
[INFO 24-12-21 20:56:00.7088 UTC quick_scorer_extended.cc:911] The binary was compiled without AVX2 support, but your CPU supports it. Enable it for faster model inference.
[INFO 24-12-21 20:56:00.7095 UTC kernel.cc:1061] Use fast generic engine


i:83


[WARNING 24-12-21 20:56:01.1484 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:01.1484 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:01.1484 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:01.7760 UTC kernel.cc:1233] Loading model from path /tmp/tmph99jbitv/model/ with prefix a48ca8989ffd4387
[INFO 24-12-21 20:56:01.7855 UTC kernel.cc:1061] Use fast generic engine


i:84


[WARNING 24-12-21 20:56:02.2219 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:02.2219 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:02.2219 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:03.0319 UTC kernel.cc:1233] Loading model from path /tmp/tmpq1z4k31j/model/ with prefix b2b699ecb0044ff3
[INFO 24-12-21 20:56:03.0486 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:56:03.0486 UTC kernel.cc:1061] Use fast generic engine


i:85


[WARNING 24-12-21 20:56:03.4923 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:03.4923 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:03.4923 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:04.0448 UTC kernel.cc:1233] Loading model from path /tmp/tmprpoeyanq/model/ with prefix 4c68f45b1ef245da
[INFO 24-12-21 20:56:04.0515 UTC kernel.cc:1061] Use fast generic engine


i:86


[WARNING 24-12-21 20:56:04.4816 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:04.4817 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:04.4817 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:05.2805 UTC kernel.cc:1233] Loading model from path /tmp/tmpbpo0ebms/model/ with prefix 80ed517e3d4d48cf
[INFO 24-12-21 20:56:05.2972 UTC kernel.cc:1061] Use fast generic engine


i:87


[WARNING 24-12-21 20:56:05.7286 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:05.7287 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:05.7287 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:06.5868 UTC kernel.cc:1233] Loading model from path /tmp/tmpyk2k2duw/model/ with prefix 978b96e68dbc4fd2
[INFO 24-12-21 20:56:06.6057 UTC kernel.cc:1061] Use fast generic engine


i:88


[WARNING 24-12-21 20:56:07.0646 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:07.0646 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:07.0646 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:07.7618 UTC kernel.cc:1233] Loading model from path /tmp/tmpoofi32nt/model/ with prefix d0c35f0b4ca24dd4
[INFO 24-12-21 20:56:07.7738 UTC kernel.cc:1061] Use fast generic engine


i:89


[WARNING 24-12-21 20:56:08.2047 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:08.2047 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:08.2048 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:08.7211 UTC kernel.cc:1233] Loading model from path /tmp/tmpb7ift5un/model/ with prefix 783e3bc0b9ed4033
[INFO 24-12-21 20:56:08.7264 UTC kernel.cc:1061] Use fast generic engine


i:90


[WARNING 24-12-21 20:56:09.1520 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:09.1521 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:09.1521 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:09.8214 UTC kernel.cc:1233] Loading model from path /tmp/tmpu005_p3r/model/ with prefix 94fe95986a96415c
[INFO 24-12-21 20:56:09.8323 UTC kernel.cc:1061] Use fast generic engine


i:91


[WARNING 24-12-21 20:56:10.2650 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:10.2651 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:10.2651 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:10.8354 UTC kernel.cc:1233] Loading model from path /tmp/tmpi2f8bry_/model/ with prefix d4de35d2c07745b8
[INFO 24-12-21 20:56:10.8426 UTC kernel.cc:1061] Use fast generic engine


i:92


[WARNING 24-12-21 20:56:12.0003 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:12.0003 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:12.0004 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:13.0071 UTC kernel.cc:1233] Loading model from path /tmp/tmpc_zyvtlg/model/ with prefix 816fb743c71c44e5
[INFO 24-12-21 20:56:13.0322 UTC kernel.cc:1061] Use fast generic engine


i:93


[WARNING 24-12-21 20:56:13.5776 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:13.5776 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:13.5777 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:14.2840 UTC kernel.cc:1233] Loading model from path /tmp/tmpuax9y5qg/model/ with prefix f37371a8874142d7
[INFO 24-12-21 20:56:14.2963 UTC abstract_model.cc:1362] Engine "GradientBoostedTreesQuickScorerExtended" built
[INFO 24-12-21 20:56:14.2964 UTC kernel.cc:1061] Use fast generic engine


i:94


[WARNING 24-12-21 20:56:14.7610 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:14.7610 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:14.7610 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:15.3386 UTC kernel.cc:1233] Loading model from path /tmp/tmphcu1qxvw/model/ with prefix 392147cb96b64598
[INFO 24-12-21 20:56:15.3465 UTC kernel.cc:1061] Use fast generic engine


i:95


[WARNING 24-12-21 20:56:15.7998 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:15.7999 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:15.7999 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:16.4348 UTC kernel.cc:1233] Loading model from path /tmp/tmphbgcyk_p/model/ with prefix 355ca5ab78164192
[INFO 24-12-21 20:56:16.4449 UTC kernel.cc:1061] Use fast generic engine


i:96


[WARNING 24-12-21 20:56:16.8997 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:16.8997 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:16.8998 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:17.5727 UTC kernel.cc:1233] Loading model from path /tmp/tmp4s5ddoq3/model/ with prefix 8d02c709290b4649
[INFO 24-12-21 20:56:17.5843 UTC kernel.cc:1061] Use fast generic engine


i:97


[WARNING 24-12-21 20:56:18.0307 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:18.0307 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:18.0307 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:18.5658 UTC kernel.cc:1233] Loading model from path /tmp/tmpmgoweekc/model/ with prefix 0654e973d2fd436e
[INFO 24-12-21 20:56:18.5715 UTC kernel.cc:1061] Use fast generic engine


i:98


[WARNING 24-12-21 20:56:19.0099 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:19.0099 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:19.0100 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:19.6148 UTC kernel.cc:1233] Loading model from path /tmp/tmpwo7gw2fl/model/ with prefix a07ee5e43c2f45b9
[INFO 24-12-21 20:56:19.6235 UTC kernel.cc:1061] Use fast generic engine


i:99


[WARNING 24-12-21 20:56:20.0713 UTC gradient_boosted_trees.cc:1840] "goss_alpha" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:20.0713 UTC gradient_boosted_trees.cc:1851] "goss_beta" set but "sampling_method" not equal to "GOSS".
[WARNING 24-12-21 20:56:20.0714 UTC gradient_boosted_trees.cc:1865] "selective_gradient_boosting_ratio" set but "sampling_method" not equal to "SELGB".
[INFO 24-12-21 20:56:20.8377 UTC kernel.cc:1233] Loading model from path /tmp/tmpbbx26cnw/model/ with prefix 5a65dae3753e4167
[INFO 24-12-21 20:56:20.8534 UTC kernel.cc:1061] Use fast generic engine


Submission exported to /kaggle/working/submission.csv


# What is next

If you want to learn more about TensorFlow Decision Forests and its advanced features, you can follow the official documentation [here](https://www.tensorflow.org/decision_forests) 